libtenzir/builtins/endpoints/ping.cpp

```diff
@@ -0,0 +1,105 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The TENZIR Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#include <tenzir/builtin_rest_endpoints.hpp>
+#include <tenzir/plugin.hpp>
+#include <tenzir/version.hpp>
+
+#include <caf/typed_event_based_actor.hpp>
+
+namespace tenzir::plugins::rest_api::ping {
+
+static auto const* SPEC_V0 = R"_(
+/ping:
+  get:
+    summary: Returns a success response
+    description: Returns a success response to indicate that the node is able to respond to requests. The response body includes the current node version.
+    responses:
+      200:
+        description: OK.
+        content:
+          application/json:
+            schema:
+              type: object
+              properties:
+                version:
+                  type: string
+                  description: The version of the responding node.
+                  example: "v2.3.0-rc3-32-g8529a6c43f"
+            example:
+              version: v2.3.0-rc3-32-g8529a6c43f
+      401:
+        description: Not authenticated.
+    )_";
+
+using ping_handler_actor
+  = typed_actor_fwd<>::extend_with<rest_handler_actor>::unwrap;
+
+struct ping_handler_state {
+  static constexpr auto name = "ping-handler";
+  ping_handler_state() = default;
+};
+
+auto ping_handler(ping_handler_actor::stateful_pointer<ping_handler_state> self)
+  -> ping_handler_actor::behavior_type {
+  return {
+    [self](atom::http_request, uint64_t,
+           const tenzir::record&) -> caf::result<rest_response> {
+      TENZIR_DEBUG("{} handles /ping request", *self);
+      auto versions = retrieve_versions();
+      TENZIR_ASSERT_CHEAP(versions.contains("Tenzir"));
+      return rest_response{record{{{"version", versions.at("Tenzir")}}}};
```

Comment:
Instead of this, you can also just use the `tenzir::version::version` string from the CMake-generated `tenzir/config.hpp` file.